interfaces/builtin/uio.go

```diff
@@ -100,6 +100,8 @@ func (iface *uioInterface) AppArmorConnectedPlug(spec *apparmor.Specification, p
 	// device node in /dev. Use AddDeduplicatedSnippet() for clarity
 	// in the resulting rules.
 	spec.AddDeduplicatedSnippet("/sys/devices/platform/**/uio/uio[0-9]** r,  # common rule for all uio connections")
+	// allow configuring industrial I/O cards UIO drivers from userspace app/libraries
+	spec.AddDeduplicatedSnippet("/sys/devices/{pci[0-9a-f]*,platform,virtual}/**/config rw,")
```

Comment:
Hm, this rule is not limited to uio devices - or am I missing something? So it grans "config" access to any device on the system(?) as written right now?

Reply:
well, this is correct. But, external IO cards can leverage various different interfaces and generally speaking paths like `/sys/class/uio/uioX/device/config`  is a symlink to `/sys/devices/{pci[0-9a-f]*,platform,virtual}/**/config`. AFAIK, if you want to access a specific path with AppArmor, you should need to add source of the path.

I can update the comment and put a clear message that, it would grant access to any device. Would that be accepted?

Reply:
I agree with @mvo5 - this regex is too generic for the `uoi` interface - would it be possible to instead implementing the symlink dereferencing within the uio interface code so that it can dynamically find these files? ie. look up and dereference all `/sys/class/uio/uioX/device/config` files and add AppArmor rules for the resulting dereferenced files?

Reply:
Hi @alexmurray, I implemented a smarter logic that dereferences `/sys/class/uio/uioX/device/config` path and adds it to the AppArmor rule. The implementation is being tested and verified.

Reply:
Thanks - do you think it might be possible to add a test for this as well?

Reply:
added test